pkg/controller/v1alpha1/localmodelnode/controller.go

```diff
@@ -133,13 +165,26 @@ func (c *LocalModelNodeReconciler) launchJob(ctx context.Context, jobName string
 	if err != nil {
 		return nil, err
 	}
-	if job != nil && reflect.DeepEqual(job.Spec.Template.Spec, dryrunJob.Spec.Template.Spec) {
+
+	// If we do not recreate job on purpose, return the existing job if it is the same as the expected job
+	if !recreateJob && job != nil && reflect.DeepEqual(job.Spec.Template.Spec, dryrunJob.Spec.Template.Spec) {
```

Comment:
would checking `jobFound` better than`job != nil` if it's checking existence of the job?